src/Schema/Element/GroupRef.php

```diff
@@ -63,6 +95,20 @@ public function getElements()
 
     public function addElement(ElementItem $element)
     {
-        throw new \Exception("Can't set the name for a ref group");
+        throw new BadMethodCallException("Can't add an element for a ref group");
+    }
+
+    /**
+    * @return GroupRef
+    */
+    public static function loadGroupRef(Group $referenced, DOMElement $node)
+    {
+        $ref = new GroupRef($referenced);
+        $ref->setDoc(SchemaReader::getDocumentation($node));
+
+        SchemaReader::maybeSetMax($ref, $node);
```

Comment:
this class (and all the other schema classes...) should not have dependencies on the DOM and on the schema reader.

helper methods should stay in helper classes. can you move the `public static` methods somewhere else?

Reply:
this was shifted onto GroupRef as part of the sprint to raise the quality of SchemaReader before I relented and split SchemaReader into several abstract classes- would be better to have a resolution to the ridiculous abstraction before moving it back?

Reply:
I think if you follow [this](https://github.com/goetas-webservices/xsd-reader/pull/18#discussion_r146490023), this code will have its natural place to be